harmonica/tests/test_legendre.py

```diff
@@ -0,0 +1,203 @@
+
+
+def full_normalization(p):
+    "Calculate the full normalization factor"
+    max_degree = p.shape[0] - 1
+    for n in range(max_degree + 1):
+        for m in range(n + 1):
+            if m == 0:
+                p[n, m] *= np.sqrt(2 * n + 1)
+            else:
+                p[n, m] *= np.sqrt(
+                    2 * (2 * n + 1) * math.factorial(n - m) / math.factorial(n + m)
+                )
+
+
+@pytest.mark.parametrize(
+    "func,norm",
+    (
+        (associated_legendre, None),
+        (associated_legendre_schmidt, schmidt_normalization),
+        (associated_legendre_full, full_normalization),
+    ),
+    ids=["unnormalized", "schmidt", "full"],
+)
+def test_associated_legendre_function_analytical(func, norm):
+    "Check if the first few degrees match analytical expressions"
+    for angle in np.linspace(0, np.pi, 360):
+        x = np.cos(angle)
+        # Analytical expression
+        p_analytical = legendre_analytical(x)
+        max_degree = p_analytical.shape[0] - 1
+        if norm is not None:
+            norm(p_analytical)
+        # Numerical calculation
+        p = np.empty((max_degree + 1, max_degree + 1))
+        func(x, max_degree, p)
+        for n in range(max_degree + 1):
+            for m in range(n + 1):
+                np.testing.assert_allclose(p_analytical[n, m], p[n, m], atol=1e-10)
+
+
+@pytest.mark.parametrize(
+    "func,deriv,norm",
+    (
+        (associated_legendre, associated_legendre_derivative, None),
+        (
+            associated_legendre_schmidt,
+            associated_legendre_schmidt_derivative,
+            schmidt_normalization,
+        ),
+        (
+            associated_legendre_full,
+            associated_legendre_full_derivative,
+            full_normalization,
+        ),
+    ),
+    ids=["unnormalized", "schmidt", "full"],
+)
+def test_associated_legendre_function_derivative_analytical(func, norm, deriv):
+    "Check if the first few degrees match analytical expressions"
+    for angle in np.linspace(0, np.pi, 360):
+        x = np.cos(angle)
+        # Analytical expression
+        dp_analytical = legendre_derivativeative_analytical(x)
+        max_degree = dp_analytical.shape[0] - 1
+        if norm is not None:
+            norm(dp_analytical)
+        # Numerical calculation
+        p = np.empty((max_degree + 1, max_degree + 1))
+        dp = np.empty((max_degree + 1, max_degree + 1))
+        func(x, max_degree, p)
+        deriv(max_degree, p, dp)
+        for n in range(max_degree + 1):
+            for m in range(n + 1):
+                np.testing.assert_allclose(
+                    dp_analytical[n, m], dp[n, m], atol=1e-10, err_msg=f"n={n}, m={m}"
+                )
+
+
+def test_associated_legengre_function_schmidt_identity():
+    "Check Schmidt normalized functions against a known identity"
+    # Higher degrees than this yield bad results
+    max_degree = 2800
```

Comment:
Using the associated legendre functions with Schmidt normalization leads to inaccurate results for degrees larger than 2800? If that's so, I think we should document this in the function itself.

Reply:
Great. Sorry I missed it.